### DISCUSSION TITLE
Does Quarto (plan to) support rendering MDX to PDFs?

### DISCUSSION THREAD

**Discussion by krishaamer** [quarto-dev/quarto-cli#6464] [Q&A]
### Description

As far as I understand MDX is currently only rendered to web?

What I'm doing currently is using conditionals based on where it's exported but it leads to repeating myself.

```
::: {.content-hidden unless-format="pdf"}
![CO2 Concentration. Photo Credit: Scripps Institution of Oceanography at UC San Diego.](./images/co2-concentration.png)
:::

{=mdx}
<Figure
  caption="CO2 Concentration"
  src={CO2}
  refURL="https://keelingcurve.ucsd.edu/"
  refTitle="Scripps Institution of Oceanography at UC San Diego"
/>

```
Thanks
Kris

**Comment by rgaiacs** [comment]
Hi @krishaamer and thanks for the message. Rendering MDX to HTML is straighforward because the MDX block is copied to the HTML and the rendering is delegated to the readers browser with JavaScript. Rendering MDX to PDF is a mine field because the MDX block must to be render to HTML with a server render (like Node) and the HTML render to LaTeX.

From the example that you provide, one solution is to use a filter (more at https://quarto.org/docs/extensions/filters.html)

**Reply by mcanouil** [reply]
Indeed and more than filter, in your case, a shortcode could help reduce of duplication of any tedious step.
